### ISSUE TITLE
When using an AWS S3 Bucket - Selecting 'Package will be downloaded directly from the feed on the execution target' on Deploy Package Step results in Feed Ui error on deploy.

### ISSUE THREAD

**Issue by Clare-Octopus** [OctopusDeploy/Issues#7974]
### Team

- [X] I've assigned a team label to this issue

### Severity

Blocking any customers who are using an AWS S3 Bucket and select the 'The package will be downloaded directly from the feed on the execution target' option on Deploy package step.

### Version

2022.2.6971

### Latest Version

I could reproduce the problem in the latest build

### What happened?

This is specifically related to AWS S3 Bucket repositories, if you are seeing it with any other repositories please let us know either via the comments section in this issue or by contacting suport@octopus.com and linking this GitHub issue.

If you have an AWS S3 Bucket repository and use it in a Deploy a Package step with the option `The package will be downloaded directly from the feed on the execution target selected` (see below image) you will get an error on deployment:

```
15:10:17   Error    |         Failed to download package xxxxx/xxxx v x.x.x from feed: ''
15:10:17   Error    |         No feed URI was specified. Please pass --feedUri https://url/to/nuget/feed
```

![image](https://user-images.githubusercontent.com/92790462/211010384-7cfd6b86-f461-460f-8229-0c4704ef99d4.png)

### Reproduction

1. Setup an AWS S3 Bucket External Feed.
2. Create a project with a 'Deploy a Package' step.
3. Due to [this bug](https://github.com/OctopusDeploy/Issues/issues/7943) you may need to enter the AWS S3 Bucket Feed ID in manually.
4. Once the feed is inputted select the package from the dropdown list of found packages.
5. Ensure `The package will be downloaded directly from the feed on the execution target selected` is ticked.
6. Save the project, create a release and deploy.
7. See the error message `No feed URI was specified`.
8. Go back into the deployment process and change the setting to the default :
`Octopus server will download the package then upload it to the tentacle`
9. Create a release and deploy, you will see it be successful.

### Error and Stacktrace

```shell
15:10:17   Error    |         Failed to download package xxxxx/xxxx v x.x.x from feed: ''
15:10:17   Error    |         No feed URI was specified. Please pass --feedUri https://url/to/nuget/feed
```

### More Information

I found this myself but it also affected this customer (internal link) - https://octopus.zendesk.com/agent/tickets/104599
Slack link to the discussion on this (internal) - https://octopusdeploy.slack.com/archives/C020P7U9Z9S/p1672850811457729
May be linked to this bug here - #7943 

I tried with a Nexus feed I had and it works with the `The package will be downloaded directly from the feed on the execution target selected` ticked so this does look to be specific to an AWS S3 Bucket feed.

### Workaround

Select the default option:

`Octopus server will download the package then upload it to the tentacle`

You should then be able to deploy the package with no issues.

![image](https://user-images.githubusercontent.com/92790462/211010855-9efe0c13-9721-43d8-8d96-b27e6ac67638.png)

**Comment by IsaacCalligeros95** [comment]
This was resolved in over in https://github.com/OctopusDeploy/Issues/issues/8188, which is a duplicate of this ticket but has the fix linked. As this is now resolved in master, this will also be back ported to 2023.3

**Comment by Clare-Octopus** [comment]
To alleviate any confusion for this specific GitHub issue fix, you will need to be using `Calamari 25.5.30-hotfix0007` or later, which is included with Octopus Server version `2023.3.13091` or above. All Octopus Cloud users are already using a Calamari version higher than this.

However, you will then run into this GitHub issue [here](https://github.com/OctopusDeploy/Issues/issues/8188), which I have unfortunately had to re-open. If you are running into this specific GitHub issue I suggest you subscribe to the GitHub issue I linked above and wait for the Octopus Server version fix number for that so you can upgrade.

I apologise for any confusion caused here. Please reach out to support@octopus.com if you have a valid, paid license and we can answer any questions you have relating to this issue.

Kind Regards,
Clare
